### PR TITLE
feat: add `s.string.phone` 

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
fixes: #200

### CODE REVIEW

**Review by favna — APPROVED** [review]

File: tests/validators/string.test.ts [hunk 1]
```diff
@@ -365,31 +365,51 @@ describe('StringValidator', () => {
...
 
-				test('GIVEN invalid date string THEN throws a ConstraintError', () => {
+			test.each(['+79919542975', '786-307-3615', '+16308520397', '+919367788755', '+916000060091', '9365706789', '936-570-6789'])(
+				'GIVEN %j THEN returns given value',
+				(input) => {
+					expect(phonePredicate.parse(input)).toBe(input);
+				}
+			);
+
+			test.each(['+1 555-555-5555 ext', '+1 (555) 555-5555', '987-123-4567 x12345', '(123) 456-7890 ext12345'])(
```
Comment (vladfrangu) [inline tests/validators/string.test.ts:404]: Why does `+1 (555) 555-5555` fail exactly? it *is* a valid number 🤔

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: tests/validators/string.test.ts (hunk 1 above)
Comment (imranbarbhuiya) [reply tests/validators/string.test.ts:404]: I used the regex fava provided. I don't have much idea of numbers with area code. Perhaps u can't use area code with country code? If that's not the case, I'll search for a better regex

Comment (favna) [reply tests/validators/string.test.ts:404]: looks like this one may be better:
```ts
^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$
```

Source: https://stackoverflow.com/a/16699507

**Comment by imranbarbhuiya** [comment]
I've updated the regex. Also changed `phone` from `getter` to a `method`. The reason is maybe we can add `allowedCountryCode`. Will try to do that in a followup pr

**Comment by vladfrangu** [comment]
Getting that quota are we? :P Yeah, sure a different PR for adding that would be fine (since it'd also need more testing)

**Review by vladfrangu — APPROVED** [review]

**Review by favna — APPROVED** [review]
sounds quite complex to do country code so you rather than me lol. But hey if you can figure it out then awesome.

**Pull request merged**